### ISSUE TITLE
Get extension ID range

### ISSUE THREAD

**Issue by AbsolutelyFreeWeb** [microsoft/AL#5246]
Is there a way to know the object range inside a codeunit ?

I would like to get min and max of my id ranges within the upgrade code-units, but have to hardcode them for now, as I don't find a function giving me them.

**Comment by atoader** [comment]
@AbsolutelyFreeWeb are you talking about getting the ID range specified in the app.json?

**Comment by AbsolutelyFreeWeb** [comment]
@atoader yes correct. Is there a way not to hard code them in the upgrade codeunit?

even a pre-compiler command should do..

**Comment by atoader** [comment]
Unfortunately, there is no way to get them at the moment. Long term, we want to move away from using IDs. You might be able to use AllObjWithCaption, AllObj or Nav App Object Metadata to get some of this information, but reflecting on the app.json is currently not supported.

**Comment by metda11** [comment]
Any Update on this? 
I would also like to check all customer apps regarding overlapping ID Range Settings.
At least I know, that they don´t have overlapping Objects :-)
Thanks

**Comment by NKarolak** [comment]
> Any Update on this? I would also like to check all customer apps regarding overlapping ID Range Settings. At least I know, that they don´t have overlapping Objects :-) Thanks

You can use command `AL: Explorer` to list all IDs within your app.

**Comment by metda11** [comment]
> > Any Update on this? I would also like to check all customer apps regarding overlapping ID Range Settings. At least I know, that they don´t have overlapping Objects :-) Thanks
> 
> You can use command `AL: Explorer` to list all IDs within your app.

I would like to use it through an API to check if the customer has Apps installed with overlapping app id ranges.
-> export all Apps with the App Range Setup from the Application, I don´t have the source code
